Wayland build fails with C++17

<!--- Please fill out this template to the best of your ability. You can always edit this issue once you have created it. -->
<!--- Read the following link before you create a new problem report: https://kodi.wiki/view/HOW-TO:Submit_a_bug_report  -->
## Bug report
### Describe the bug
## Actual Behavior
Here is a clear and concise description of what the problem is:

````
[ 4069s] [ 97%] Building CXX object build/windowing/wayland/CMakeFiles/windowing_WAYLAND.dir/WindowDecorator.cpp.o
[ 4069s] cd /home/abuild/rpmbuild/BUILD/kodi-18.1.20190320T070326.b0e84f1c8ef/build/build/windowing/wayland && /usr/bin/c++   -I/home/abuild/rpmbuild/BUILD/kodi-18.1.20190320T070326.b0e84f1c
8ef/build -I/home/abuild/rpmbuild/BUILD/kodi-18.1.20190320T070326.b0e84f1c8ef -I/home/abuild/rpmbuild/BUILD/kodi-18.1.20190320T070326.b0e84f1c8ef/lib -I/home/abuild/rpmbuild/BUILD/kodi-18.1.
20190320T070326.b0e84f1c8ef/lib/gtest/include -I/home/abuild/rpmbuild/BUILD/kodi-18.1.20190320T070326.b0e84f1c8ef/xbmc -I/home/abuild/rpmbuild/BUILD/kodi-18.1.20190320T070326.b0e84f1c8ef/xbm
c/platform/linux -I/home/abuild/rpmbuild/BUILD/kodi-18.1.20190320T070326.b0e84f1c8ef/xbmc/cores/VideoPlayer -I/home/abuild/rpmbuild/BUILD/kodi-18.1.20190320T070326.b0e84f1c8ef/build/build -I
/usr/include/dbus-1.0 -I/usr/lib64/dbus-1.0/include -I/usr/include/python2.7 -I/usr/include/samba-4.0 -I/usr/include/libxml2 -I/home/abuild/rpmbuild/BUILD/kodi-18.1.20190320T070326.b0e84f1c8
ef/build/build/cpluff/include -I/usr/include/ffmpeg -I/home/abuild/rpmbuild/BUILD/kodi-18.1.20190320T070326.b0e84f1c8ef/build/build/cores/RetroPlayer/messages -I/usr/include/freetype2 -I/usr
/include/lzo -I/usr/include/libdrm -I/usr/include/libxkbcommon  -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -
DNDEBUG -DWL_EGL_PLATFORM=1 -DPLATFORM_SETTINGS_FILE=wayland.xml -O3 -DNDEBUG -s   -D_LINUX -DTARGET_POSIX -DTARGET_LINUX -D_GNU_SOURCE -DHAVE_LINUX_MEMFD=1 -DHAVE_MKOSTEMP=1 -D__STDC_CONSTA
NT_MACROS -D_FILE_DEFINED -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -DHAS_LINUX_NETWORK -DHAVE_STD__U16_STRING=1 -DHAVE_STD__U32_STRING=1 -DHAVE_CHAR16_T=1 -DHAVE_CHAR32_T=1 -DHAVE_STDINT
_H=1 -DHAS_BUILTIN_SYNC_ADD_AND_FETCH=1 -DHAS_BUILTIN_SYNC_SUB_AND_FETCH=1 -DHAS_BUILTIN_SYNC_VAL_COMPARE_AND_SWAP=1 -DHAVE_INOTIFY=1 -DHAVE_POSIX_FADVISE=1 -DHAVE_LOCALTIME_R=1 -DHAVE_INTTY
PES_H=1 -DHAS_ALSA=1 -DHAS_AVAHI=1 -DHAS_ZEROCONF=1 -DHAVE_LIBBLURAY=1 -DHAVE_LIBBLURAY_BDJ=1 -DHAVE_LIBCEC=1 -DHAS_DBUS=1 -DHAVE_LCMS2=1 -DHAS_LIRC=1 -DHAS_WEB_SERVER=1 -DHAS_WEB_INTERFACE=
1 -DHAS_FILESYSTEM_NFS=1 -DHAS_PULSEAUDIO=1 -DHAS_PYTHON=1 -DHAS_FILESYSTEM_SMB=1 -DHAS_SNDIO=1 -DHAVE_LIBUDEV=1 -DHAVE_LIBXSLT=1 -DHAVE_LIBVA=1 -DHAS_AIRPLAY=1 -DFFMPEG_VER_SHA=\"4.0\" -I/u
sr/include/fribidi -DHAS_EGL=1 -DHAVE_WAYLAND=1 -I/usr/include/wayland -DHAS_GLES=3 -DHAS_MYSQL=1 -DHAS_UPNP=1 -DHAS_DVD_DRIVE -DHAS_CDDA_RIPPER -DHAS_AIRTUNES=1 -DBIN_INSTALL_PATH=\"/usr/lib64/kodi\" -DINSTALL_PATH=\"/usr/share/kodi\" -std=c++17 -o CMakeFiles/windowing_WAYLAND.dir/WindowDecorator.cpp.o -c /home/abuild/rpmbuild/BUILD/kodi-18.1.20190320T070326.b0e84f1c8ef/xbmc/w
indowing/wayland/WindowDecorator.cpp
[ 4070s] In file included from /usr/include/c++/8/set:60,
[ 4070s]                  from /home/abuild/rpmbuild/BUILD/kodi-18.1.20190320T070326.b0e84f1c8ef/xbmc/windowing/wayland/WindowDecorator.h:13,
[ 4070s]                  from /home/abuild/rpmbuild/BUILD/kodi-18.1.20190320T070326.b0e84f1c8ef/xbmc/windowing/wayland/WindowDecorator.cpp:9:
[ 4070s] /usr/include/c++/8/bits/stl_tree.h: In instantiation of 'class std::_Rb_tree<wayland::buffer_t, wayland::buffer_t, std::_Identity<wayland::buffer_t>, KODI::WINDOWING::WAYLAND::WaylandCPtrCompare, std::allocator<wayland::buffer_t> >':
[ 4070s] /usr/include/c++/8/bits/stl_set.h:133:17:   required from 'class std::set<wayland::buffer_t, KODI::WINDOWING::WAYLAND::WaylandCPtrCompare>'
[ 4070s] /home/abuild/rpmbuild/BUILD/kodi-18.1.20190320T070326.b0e84f1c8ef/xbmc/windowing/wayland/WindowDecorator.h:209:51:   required from here
[ 4070s] /usr/include/c++/8/bits/stl_tree.h:457:21: error: static assertion failed: comparison object must be invocable as const
--
[ 4075s] [ 97%] Building CXX object build/windowing/wayland/CMakeFiles/windowing_WAYLAND.dir/WinSystemWayland.cpp.o
[ 4075s] cd /home/abuild/rpmbuild/BUILD/kodi-18.1.20190320T070326.b0e84f1c8ef/build/build/windowing/wayland && /usr/bin/c++   -I/home/abuild/rpmbuild/BUILD/kodi-18.1.20190320T070326.b0e84f1c8ef/build -I/home/abuild/rpmbuild/BUILD/kodi-18.1.20190320T070326.b0e84f1c8ef -I/home/abuild/rpmbuild/BUILD/kodi-18.1.20190320T070326.b0e84f1c8ef/lib -I/home/abuild/rpmbuild/BUILD/kodi-18.1.20190320T070326.b0e84f1c8ef/lib/gtest/include -I/home/abuild/rpmbuild/BUILD/kodi-18.1.20190320T070326.b0e84f1c8ef/xbmc -I/home/abuild/rpmbuild/BUILD/kodi-18.1.20190320T070326.b0e84f1c8ef/xbmc/platform/linux -I/home/abuild/rpmbuild/BUILD/kodi-18.1.20190320T070326.b0e84f1c8ef/xbmc/cores/VideoPlayer -I/home/abuild/rpmbuild/BUILD/kodi-18.1.20190320T070326.b0e84f1c8ef/build/build -I/usr/include/dbus-1.0 -I/usr/lib64/dbus-1.0/include -I/usr/include/python2.7 -I/usr/include/samba-4.0 -I/usr/include/libxml2 -I/home/abuild/rpmbuild/BUILD/kodi-18.1.20190320T070326.b0e84f1c8ef/build/build/cpluff/include -I/usr/include/ffmpeg -I/home/abuild/rpmbuild/BUILD/kodi-18.1.20190320T070326.b0e84f1c8ef/build/build/cores/RetroPlayer/messages -I/usr/include/freetype2 -I/usr/include/lzo -I/usr/include/libdrm -I/usr/include/libxkbcommon  -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -DNDEBUG -DWL_EGL_PLATFORM=1 -DPLATFORM_SETTINGS_FILE=wayland.xml -O3 -DNDEBUG -s   -D_LINUX -DTARGET_POSIX -DTARGET_LINUX -D_GNU_SOURCE -DHAVE_LINUX_MEMFD=1 -DHAVE_MKOSTEMP=1 -D__STDC_CONSTANT_MACROS -D_FILE_DEFINED -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -DHAS_LINUX_NETWORK -DHAVE_STD__U16_STRING=1 -DHAVE_STD__U32_STRING=1 -DHAVE_CHAR16_T=1 -DHAVE_CHAR32_T=1 -DHAVE_STDINT_H=1 -DHAS_BUILTIN_SYNC_ADD_AND_FETCH=1 -DHAS_BUILTIN_SYNC_SUB_AND_FETCH=1 -DHAS_BUILTIN_SYNC_VAL_COMPARE_AND_SWAP=1 -DHAVE_INOTIFY=1 -DHAVE_POSIX_FADVISE=1 -DHAVE_LOCALTIME_R=1 -DHAVE_INTTYPES_H=1 -DHAS_ALSA=1 -DHAS_AVAHI=1 -DHAS_ZEROCONF=1 -DHAVE_LIBBLURAY=1 -DHAVE_LIBBLURAY_BDJ=1 -DHAVE_LIBCEC=1 -DHAS_DBUS=1 -DHAVE_LCMS2=1 -DHAS_LIRC=1 -DHAS_WEB_SERVER=1 -DHAS_WEB_INTERFACE=1 -DHAS_FILESYSTEM_NFS=1 -DHAS_PULSEAUDIO=1 -DHAS_PYTHON=1 -DHAS_FILESYSTEM_SMB=1 -DHAS_SNDIO=1 -DHAVE_LIBUDEV=1 -DHAVE_LIBXSLT=1 -DHAVE_LIBVA=1 -DHAS_AIRPLAY=1 -DFFMPEG_VER_SHA=\"4.0\" -I/usr/include/fribidi -DHAS_EGL=1 -DHAVE_WAYLAND=1 -I/usr/include/wayland -DHAS_GLES=3 -DHAS_MYSQL=1 -DHAS_UPNP=1 -DHAS_DVD_DRIVE -DHAS_CDDA_RIPPER -DHAS_AIRTUNES=1 -DBIN_INSTALL_PATH=\"/usr/lib64/kodi\" -DINSTALL_PATH=\"/usr/share/kodi\" -std=c++17 -o CMakeFiles/windowing_WAYLAND.dir/WinSystemWayland.cpp.o -c /home/abuild/rpmbuild/BUILD/kodi-18.1.20190320T070326.b0e84f1c8ef/xbmc/windowing/wayland/WinSystemWayland.cpp
[ 4077s] In file included from /usr/include/c++/8/map:60,
[ 4077s]                  from /home/abuild/rpmbuild/BUILD/kodi-18.1.20190320T070326.b0e84f1c8ef/xbmc/windowing/wayland/WinSystemWayland.h:16,
[ 4077s]                  from /home/abuild/rpmbuild/BUILD/kodi-18.1.20190320T070326.b0e84f1c8ef/xbmc/windowing/wayland/WinSystemWayland.cpp:9:
[ 4077s] /usr/include/c++/8/bits/stl_tree.h: In instantiation of 'class std::_Rb_tree<wayland::buffer_t, wayland::buffer_t, std::_Identity<wayland::buffer_t>, KODI::WINDOWING::WAYLAND::WaylandCPtrCompare, std::allocator<wayland::buffer_t> >':
[ 4077s] /usr/include/c++/8/bits/stl_set.h:133:17:   required from 'class std::set<wayland::buffer_t, KODI::WINDOWING::WAYLAND::WaylandCPtrCompare>'
[ 4077s] /home/abuild/rpmbuild/BUILD/kodi-18.1.20190320T070326.b0e84f1c8ef/xbmc/windowing/wayland/WindowDecorator.h:209:51:   required from here
[ 4077s] /usr/include/c++/8/bits/stl_tree.h:457:21: error: static assertion failed: comparison object must be invocable as const
````
### To Reproduce
Steps to reproduce the behavior:
1.  append `-DCORE_PLATFORM_NAME=wayland -DWAYLAND_RENDER_SYSTEM=gles` to `cmake`
2. run make VERBOSE=1

Note that `CMAKE_CXX_STANDARD` was set to `17` because headers from crossguid.git/master require it.

Used Operating system:
- [ ] Android
 - [ ] iOS
 - [x ] Linux
 - [ ] OSX
 - [ ] Raspberry-Pi
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: openSUSE Tumbleweed
 - Kodi version: b0e84f1c8ef

--- a/xbmc/windowing/wayland/Util.h
+++ b/xbmc/windowing/wayland/Util.h
@@ -22,7 +22,7 @@ namespace WAYLAND
 
 struct WaylandCPtrCompare
 {
-  bool operator()(wayland::proxy_t const& p1, wayland::proxy_t const& p2)
+  bool operator()(wayland::proxy_t const& p1, wayland::proxy_t const& p2) const
   {
     return p1.c_ptr() < p2.c_ptr();
   }

